src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,454 @@
+      case 'large':
+        if (children) {
+          return [0, theme.spacings.xxlarge];
+        }
+        return [0, theme.spacings.small];
+      default:
+        if (children) {
+          return [0, theme.spacings.xxlarge];
+        }
+        return [0, 0.75]; //TODO: horizontal padding not on scale of 8
+    }
+  },
+  iconSize({ size, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (children) {
+          return 'xsmall';
+        }
+        return 'small';
+      case 'small':
+        if (children) {
+          return 'small';
+        }
+        return 'medium';
+      case 'medium':
+        if (children) {
+          return 'medium';
+        }
+        return 'large';
+      case 'large':
+        if (children) {
+          return 'medium';
+        }
+        return 'large';
+      default:
+        if (children) {
+          return 'medium';
+        }
+        return 'large';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+      default:
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'small':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'medium':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+};
+
+const StyledButton = styled.button`
+  &&& {
+    background-color: ${styles.backgroundColor};
+    border-radius: ${(props) => props.theme.spacings.xxsmall};
+    border: ${styles.border};
+    width: ${(props) => (props.block ? '100%' : '')};
+    outline: none;
```

Comment:
Add `cursor: pointer;`

Reply:
Add `pointer-events` property, set as `none` for `disabled` state.

Reply:
added

Reply:
Noice. 💯

Reply:
Actually, for `cursor` as well the designers might want to not have `pointer`.
Can you please check once on Figma or someone from the design team?

Reply:
confirmed from design.